### PR TITLE
merge branch-23.06 to main branch

### PR DESCRIPTION
**Pull request by nvliyuan**
Signed-off-by: liyuan <yuali@nvidia.com>
merge branch-23.06 to main branch

**please merge the pr rather than squash and merge**

### CODE REVIEW

**Review by pxLi — APPROVED** [review]
**NOTE:** must be merged as `create a merge commit`, thanks

**Review by pxLi — CHANGES_REQUESTED** [review]
I saw many version to 23.02 for plugin, is this intentional?

File: examples/UDF-Examples/Spark-cuSpatial/README.md [hunk 1]
```diff
@@ -84,7 +85,7 @@ Note: The docker env is just for building the jar, not for running the applicati
 1. Set up [a standalone cluster](/docs/get-started/xgboost-examples/on-prem-cluster/standalone-scala.md) of Spark. Make sure the conda/lib is included in LD_LIBRARY_PATH, so that spark executors can load libcuspatial.so.
 
 2. Download Spark RAPIDS JAR
-   * [Spark RAPIDS JAR v23.04.0](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/23.04.0/rapids-4-spark_2.12-23.04.0.jar) or above
+   * [Spark RAPIDS JAR v23.02.0](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/23.02.0/rapids-4-spark_2.12-23.02.0.jar) or above
```
Comment (pxLi) [inline examples/UDF-Examples/Spark-cuSpatial/README.md:88]: I understand some example case does not require version, bump up but for released plugin: elder version may not work any more.
did we verify this?

**Review by pxLi — APPROVED** [review]
LGTM if plugin ver of cuspatial example is good~

**Review by NvTimLiu — APPROVED** [review]

**Pull request merged**